### PR TITLE
Nm sentence length

### PR DESCRIPTION
**Pull request by nickmcmahon01**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt [hunk 1]
```diff
@@ -177,12 +177,16 @@ data class Sentence @JsonCreator constructor(
   val terminationDate: LocalDate?,
 
   @JsonProperty("sentenceType")
-  val sentenceType: SentenceType,
-)
+  val sentenceType: KeyValue,
```
Comment (nickmcmahon01) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt:180]: Sentence Type like most things in community-api is really just a keyvaluepair

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt [hunk 2]
```diff
@@ -209,9 +213,7 @@ data class DeliusAssessments @JsonCreator constructor(
 
 data class KeyValue @JsonCreator constructor(
   @JsonProperty("code")
-  val code: String,
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt:212]: ❤️

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt [hunk 3]
```diff
@@ -111,17 +111,11 @@ class ProtectLevelCalculator(
...
+        val violenceArsonPoints = if (calculationVersion >= 1.1) getArsonOrViolencePoints(convictions) else 0
 
-        val tenMonthsPlus = when {
-          hasTenMonthSentencePlus(convictions) -> if (enableFemaleTenMonthsPlusCheck) 2 else 0
-          else -> 0
-        }
+        val tenMonthsPlusOrIndeterminatePoints = if (calculationVersion >= 1.1) getSentenceLengthPoints(convictions) else 0
 
-        additionalFactorsPoints + breachRecallPoints + violenceArson + tenMonthsPlus
+        additionalFactorsPoints + breachRecallPoints + violenceArsonPoints + tenMonthsPlusOrIndeterminatePoints
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt:118]: This section is nicer now

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt [hunk 4]
```diff
@@ -149,13 +143,20 @@ class ProtectLevelCalculator(
...
+    if (convictions.flatMap { it.offences }
       .map { it.offenceDetail }.any {
         OFFENCE_CODES.contains(it.mainCategoryCode)
       }
+    ) 2 else 0
+
+  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int {
+    val sentences = convictions.map { it.sentence }
+    val longerThanTenMonths = sentences.any { it.startDate != null && it.expectedSentenceEndDate != null && Period.between(it.startDate, it.expectedSentenceEndDate).months >= 10 }
+    val indeterminateSentence = sentences.any { it.latestCourtAppearanceOutcome?.code == "303" }
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculator.kt:156]: There is definitely no need to change this, but again here I feel that Sentence should either be suffixed to both vals or neither (neither is more readable in my view)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculatorTest.kt [hunk 5]
```diff
@@ -24,7 +24,6 @@ import uk.gov.justice.digital.hmpps.hmppstier.client.Offender
 import uk.gov.justice.digital.hmpps.hmppstier.client.OffenderAssessment
 import uk.gov.justice.digital.hmpps.hmppstier.client.Registration
 import uk.gov.justice.digital.hmpps.hmppstier.client.Sentence
-import uk.gov.justice.digital.hmpps.hmppstier.client.SentenceType
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/ProtectLevelCalculatorTest.kt:27]: So many "Not Used"

**Review by markreesmoj — CHANGES_REQUESTED** [review]
Do we need to check whether the sentence is custodial? From the ticket it looks like we do but I can't see it here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt (hunk 2 above)
Comment (markreesmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/client/CommunityApiClient.kt:212]: Glad to see the removal of the unused description

**Comment by nickmcmahon01** [comment]
> Do we need to check whether the sentence is custodial? From the ticket it looks like we do but I can't see it here

no that's a good spot, I'd made an assumption about the data.

**Pull request merged**